app/server/lib/coreCreator.ts

```diff
@@ -26,24 +25,35 @@ export class CoreCreate extends BaseCreate {
     super('core', storage);
   }
 
-  public override getAttachmentStoreOptions(): ICreateAttachmentStoreOptions[] {
-    return [
-      {
-        name: 'minio',
-        isAvailable: async () => checkMinIOExternalStorage() !== undefined,
+  public override getAttachmentStoreOptions() {
+    return {
+      // 'snapshots' provider uses the ExternalStorage provider set up for doc snapshots for attachments
+      snapshots: {
+        name: 'snapshots',
+        isAvailable: async () => {
+          try {
+            const storage = this.ExternalStorage('attachments', '');
+            return storage ? storageSupportsAttachments(storage) : false;
+          } catch {
+            // Need to catch exceptions, as some storages won't support the 'attachments' purpose.
```

Comment:
Is there a way to check for the expected error here if attachments not supported? So more serious errors don't get swallowed?

Reply:
No objection from me - we've got a helper function that basically already does this try / catch block that's used at a higher level.